aws/resource_aws_storagegateway_stored_iscsi_volume.go

```diff
@@ -0,0 +1,270 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/storagegateway"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsStorageGatewayStoredIscsiVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsStorageGatewayStoredIscsiVolumeCreate,
+		Read:   resourceAwsStorageGatewayStoredIscsiVolumeRead,
+		Update: resourceAwsStorageGatewayStoredIscsiVolumeUpdate,
+		Delete: resourceAwsStorageGatewayStoredIscsiVolumeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"disk_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"gateway_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"target_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"preserve_existing_data": {
+				Type:     schema.TypeBool,
+				Required: true,
+				ForceNew: true,
+			},
+			"kms_encrypted": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"kms_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+				RequiredWith: []string{"kms_encrypted"},
+			},
+			// Poor API naming: this accepts the IP address of the network interface
+			"network_interface_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network_interface_port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"snapshot_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"chap_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"lun_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"target_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"volume_arn": {
```

Comment:
Do we need both `arn` and `volume_arn`? They are always set to the same value.

Reply:
good point, only for api parity but ill remove it and in can be added later if needed